src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx

```diff
@@ -413,6 +429,37 @@ class ExplorerButtonGroup extends React.Component {
     });
   }
 
+  // This is a generic way to export a PFB to a third party, deprecating
+  // exportToTerra and exportToSevenBridges.
+  // See docs/export-pfb-to-url.md
+  // Code flow (it's confusing):
+  // 1. User clicks a 'export-pfb-to-url' button
+  // 2. Store target URL as state.targetURLTemplate
+  // 2. Kick off a PFB export job with this.exportToPFB()
+  // 3. componentDidUpdate polls the exportToPFB job status, when complete it will call
+  //    sendPFBToURL(this.state.targetURLTemplate)
+  exportPFBToURL = (targetURLTemplate) => () => {
+    this.setState({
+      exportingPFBToURL: true,
+      targetURLTemplate,
+    }, () => {
+      this.exportToPFB();
+    });
+  }
+
+  sendPFBToURL = (targetURLTemplate, presignedURL) => {
+    const signedURL = encodeURIComponent(presignedURL);
+    // the PFB export target URL is a template URL that should have a {{PRESIGNED_URL}} template
+    // variable in it.
+    const PRESIGNED_URL_TEMPLATE_VARIABLE = '{{PRESIGNED_URL}}';
+    const targetURLIsBad = this.state.targetURLTemplate.indexOf(PRESIGNED_URL_TEMPLATE_VARIABLE) === -1;
+    if (targetURLIsBad) {
+      throw new Error(`Misconfiguration error! An \`export-pfb-to-url\` button has a bad \`targetURLTemplate\` property. The string \`{{PRESIGNED_URL}}\` must appear in the \`targetURLTemplate\` property. Bad \`targetURLTemplate\`: ${this.state.targetURLTemplate}`);
```

Comment:
maybe better to check and throw this error in the same time when loading the button (in the same place that check if `targetURLTemplate` exists)?

Reply:
Good idea, I'll make the change.